### ISSUE TITLE
Function Signature of `_resource_apply_gradients` and sparse equivalents in abstract `Optimizer`

### ISSUE THREAD

**Issue by FelixBenning** [tensorflow/tensorflow#50219]
## URL(s) with the issue:

https://github.com/tensorflow/tensorflow/blob/v2.5.0/tensorflow/python/keras/optimizer_v2/optimizer_v2.py#L1237-L1298

## Description of issue (what needs changing):

It appears that the function signature is out of date: The second argument is called `handle` and the description indicates that the function expects a `resource` variable pointing to the variable that needs updating, but if we look at the [SGD Optimizer](https://github.com/tensorflow/tensorflow/blob/v2.5.0/tensorflow/python/keras/optimizer_v2/gradient_descent.py#L129-L149) for example it appears that in reality the variable itself is passed to this function. This problem is something I could fix myself in a pull request.

The second part I am not clear on: What is the return value supposed to be? [cf. StackOverflow Question](https://stackoverflow.com/questions/67902325/what-type-should-the-method-resource-apply-gradient-from-tensorflow-keras-op). In particular I do not know how to construct an operation without using the inbuilt functions used in the provided implementations. Which is an issue if I want to create a custom optimizer.

I am thinking that 

```python
return tf.raw_ops.ResourceApplyGradientDescent(
        var=var.handle,
        alpha=coefficients["learning_rate"],
        delta=grad,
        use_locking=self._use_locking,
)
```

might be equivalent to

```python
return var.assign_sub(
    delta=coefficients["learning_rate"]*grad,
    use_locking=self._use_locking,
    read_value=False
)
```
but I am not sure and I can not find any documentation on that

**Comment by tilakrayal** [comment]
@FelixBenning ,

We see that the[ issue template ](https://github.com/tensorflow/tensorflow/issues/new/choose)has not been filled, could you please do so as it helps us analyse the issue.Thanks!
